templates/content.macros.j2

```diff
@@ -318,3 +317,23 @@
     {{ caller() }}
   </button>
 {% endmacro %}
+
+
+{#
+  Project Links
+  -------------
+#}
+{%- macro project_link(
+  page,
+  allow_external=false,
+  allow_fallback=true
+) -%}
+  {%- set project = page.config.project[0] if page.config.project else none -%}
+  {%- set has_page = true if (project.owner or project.case_study) else false -%}
+  {%- set fallback = '/open-source/' if page.config.contributors else '/work/' -%}
+  {%- set fallback = fallback if allow_fallback else none -%}
+  {%- set local = '/' if (page.slug == 'index') else ['/', page.slug|replace('/index', ''), '/']|join -%}
```

Comment:
@jgerigmeyer I know you were moving things towards…

```
{% set url = ['/', page.slug[:-6] if page.slug.endswith('/index') else page.slug, '/']|join %}
```

But I'm not sure why. It looks more fragile to me, but if there's a reason for it, we only need to change it in one place now.

Reply:
I'm not sure why it looks more fragile to you? I don't like generic `|replace` calls, when really we only want to replace it if it's the end of the slug.